shared/utils/relabel-ids.py

```diff
@@ -187,19 +187,21 @@ def _add_cce_id_refs_to_oval_checks(self, idmappingdict):
                 continue
 
             xccdfcceid = idmappingdict[ovalid]
-            if not ssgcommon.cce_is_valid(xccdfcceid):
-                msg = "The CCE ID {0} is not in valid form.".format(xccdfcceid)
-                raise ssgcommon.SSGError(msg)
-
-            # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
-            # after <description> element of specific OVAL check
-            ccerefelem = ssgcommon.ElementTree.Element(
-                'reference', ref_id=xccdfcceid, source="CCE")
-            ovaldesc.addnext(ccerefelem)
-            # Sanity check if appending succeeded
-            if ccerefelem.getprevious() is not ovaldesc:
-                msg = "Failed to add CCE ID to {0}.".format(ovalid)
-                raise ssgcommon.SSGError(msg)
+            if ssgcommon.cce_is_valid(xccdfcceid):
+                # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
+                # after <description> element of specific OVAL check
+                ccerefelem = ssgcommon.ElementTree.Element(
+                    'reference', ref_id=xccdfcceid, source="CCE")
+                metadata = rule.find(".//{%s}metadata" % self.CHECK_NAMESPACE)
+                metadata.append(ccerefelem)
+                # Sanity check if appending succeeded
+                cce_ref = ""
```

Comment:
you reset cce_ref to empty string but then set it to None in the loop in case it's not defined. choose just one

Reply:
@mpreisler a little confused here. In a previous comment, you say that it is dangerous that cce_ref is never reset, so what should be done here?

Reply:
either reset cce_ref to None or set it to "" in the code below. Don't use None and "" interchangeably, they are different. In this case it's fine and doesn't do anything bad but it's confusing to read.

Reply:
Thanks @mpreisler that makes sense. Set to None.